h2o-r/h2o-package/R/explain.R

```diff
@@ -2425,83 +2427,100 @@ h2o.ice_plot <- function(model,
+          pdp[[col_name]] <- as.factor(pdp[[col_name]])
+      }
+      pdp[["text"]] <- paste0(
       "Partial Depencence \n",
       "Feature Value: ", pdp[[col_name]], "\n",
       "Mean Response: ", pdp[["mean_response"]], "\n"
-    )
-    y_range <- range(results$mean_response)
+      )
+    }
 
-    p <- ggplot2::ggplot(ggplot2::aes(x = .data[[col_name]],
-                                      y = .data$mean_response,
-                                      color = .data$name,
-                                      text = .data$text),
-                         data = results) +
-      stat_count_or_bin(!is.numeric(newdata[[column]]),
-                        ggplot2::aes(x = .data[[col_name]], y = (.data$..count.. / max(.data$..count..)) * diff(y_range) / 1.61),
-                        position = ggplot2::position_nudge(y = y_range[[1]] - 0.05 * diff(y_range)), alpha = 0.2,
-                        inherit.aes = FALSE, data = as.data.frame(newdata[[column]])) +
-      geom_point_or_line(!is.numeric(newdata[[column]]), ggplot2::aes(group = .data$name)) +
-      geom_point_or_line(!is.numeric(newdata[[column]]),
-                         if (is.factor(pdp[[col_name]])) {
-                           ggplot2::aes(shape = "Partial Dependence", group = "Partial Dependence")
-                         } else {
-                           ggplot2::aes(linetype = "Partial Dependence", group = "Partial Dependence")
-                         },
-                         data = pdp, color = "black"
-      ) +
-      ggplot2::geom_rug(ggplot2::aes(x = .data[[col_name]], y = NULL, text = NULL),
-                        sides = "b", alpha = 0.1, color = "black",
-                        data = stats::setNames(as.data.frame(newdata[[column]]), col_name)
-      ) +
-      ggplot2::labs(y = "Response", title = sprintf(
-        "Individual Conditional Expectations on \"%s\"%s\nfor Model: \"%s\"", col_name,
-        if (is.null(target)) {
-          ""
+    q <- ggplot2::ggplot(ggplot2::aes(x = .data[[col_name]],
+    y = .data$mean_response,
+    color = .data$name,
+    text = .data$text),
+    data = results)
+    histogram <-  stat_count_or_bin(!is.numeric(newdata[[column]]),
+    ggplot2::aes(x = .data[[col_name]], y = (.data$..count.. / max(.data$..count..)) * diff(y_range) / 1.61),
+    position = ggplot2::position_nudge(y = y_range[[1]] - 0.05 * diff(y_range)), alpha = 0.2,
+    inherit.aes = FALSE, data = as.data.frame(newdata[[column]]))
+    rug_part <- ggplot2::geom_rug(ggplot2::aes(x = .data[[col_name]], y = NULL, text = NULL),
+      sides = "b", alpha = 0.1, color = "black",
+      data = stats::setNames(as.data.frame(newdata[[column]]), col_name)
+    )
+    plot_name <-  ggplot2::labs(y = "Response", title = sprintf(
+    "Individual Conditional Expectations on \"%s\"%s\nfor Model: \"%s\"", col_name,
+    if (is.null(target)) {
+        ""
+    } else {
+        sprintf(" with Target = \"%s\"", target)
+    },
+    model@model_id
+    ))
```

Comment:
Looks like indentation got broken here. Could you reformat the code so it's easier to read and edit in the future?